### DISCUSSION TITLE
Mistune vs other Markdown libraries?

### DISCUSSION THREAD

**Discussion by irskep** [cu/silicon#23] [General]
How conscious was the choice to use Mistune? I ask because the [markdown library has an extension to add anchors to headings](https://python-markdown.github.io/extensions/toc/), but Mistune does not.

**Comment by cu** [comment]
Oh it's quite a long story. :) I spent the better part of a week or two just looking at all of the Python markdown renderers that existed. At one point, I even had a Jupyter notebook to compare them all side-by-side. It's been a couple of years now but if I had to guess, I would say that Mistune:

- Had all of the syntax that I wanted
- Was smaller, simpler, and faster than most others
- Was fairly straightforward to extend to add [various features](https://github.com/cu/silicon/blob/master/silicon/render_md.py)
- Does not bring in any other dependencies

I did look into writing a renderer to add anchors to headings at some point. As I recall, this meant duplicating most of the Mistune header renderer code, but maybe I'm misremembering.

**Comment by irskep** [comment]
I wonder if the best thing would be to contribute anchor links in headers directly to Mistune.
